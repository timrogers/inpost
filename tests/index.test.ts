import nock from 'nock';

import {
  findLocationsByPostcode,
  getAvailabilityForLocation,
  getLocation,
} from '../src/index';
import pointsFixture from './fixtures/points.json';
import pointsErrorFixture from './fixtures/points-error.json';
import pointFixture from './fixtures/point.json';
import pointErrorFixture from './fixtures/point-error.json';
import availabilityFixture from './fixtures/availability.json';
import availabilityErrorFixture from './fixtures/availability-error.json';

describe('findLocationsByPostcode', () => {
  beforeEach(() => {
    nock.disableNetConnect();
  });

  test('returns a list of locations', async () => {
    nock('https://api-uk-points.easypack24.net')
      .get('/v1/points')
      .query({
        relative_post_code: 'SW1A 1AA',
        max_distance: '16000',
        status: 'Operating',
        limit: '10',
      })
      .reply(200, pointsFixture);

    const locations = await findLocationsByPostcode('SW1A 1AA');

    expect(locations).toMatchSnapshot();
  });

  test('throws an error for a non-200 response', async () => {
    nock('https://api-uk-points.easypack24.net')
      .get('/v1/points')
      .query({
        relative_post_code: 'SW1A',
        max_distance: '16000',
        status: 'Operating',
        limit: '10',
      })
      .reply(400, pointsErrorFixture);

    expect.assertions(1);
    return expect(findLocationsByPostcode('SW1A')).rejects.toMatchSnapshot();
  });
});

describe('getLocation', () => {
  beforeEach(() => {
    nock.disableNetConnect();
  });

  test('returns the location', async () => {
    nock('https://api-uk-points.easypack24.net')
      .get('/v1/points/UK00000756')
      .reply(200, pointFixture);

    const location = await getLocation('UK00000756');

    expect(location).toMatchSnapshot();
  });

  test('throws an error for a non-200 response', async () => {
    nock('https://api-uk-points.easypack24.net')
      .get('/v1/points/UK0000075')
      .reply(400, pointErrorFixture);

    expect.assertions(1);
    return expect(getLocation('UK0000075')).rejects.toMatchSnapshot();
  });
});

describe('getAvailabilityForLocation', () => {
  test('returns availability data for a location', async () => {
    nock('https://api.inpost247.uk')
      .get('/locker-capacity')
      .query({
        apm: 'UK00058679',
      })
      .reply(200, availabilityFixture);

    const locations = await getAvailabilityForLocation('UK00058679');

    expect(locations).toMatchSnapshot();
  });

  test('throws an error for a non-200 response', async () => {
    nock('https://api.inpost247.uk')
      .get('/locker-capacity')
      .query({
        apm: 'UK000586',
      })
      .reply(404, availabilityErrorFixture);

    expect.assertions(1);
    return expect(getAvailabilityForLocation('UK000586')).rejects.toMatchSnapshot();
  });
});
