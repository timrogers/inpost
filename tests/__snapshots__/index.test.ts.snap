// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findLocationsByPostcode returns a list of locations 1`] = `
[
  {
    "id": "UKUPM00001",
    "latitude": 51.553,
    "longitude": 0.2467,
    "name": "BP Tey Service Station",
  },
  {
    "id": "UK00058679",
    "latitude": 51.57831,
    "longitude": 0.23649,
    "name": "Essex Wines",
  },
  {
    "id": "UK00013780",
    "latitude": 51.56347,
    "longitude": 0.21914,
    "name": "SAINSBURY'S Hornchurch",
  },
  {
    "id": "UK00061711",
    "latitude": 51.5686,
    "longitude": 0.21954,
    "name": "Planet Stores",
  },
  {
    "id": "UKLON00433",
    "latitude": 51.58087,
    "longitude": 0.23589,
    "name": "BP Hornchurch",
  },
  {
    "id": "UK00021042",
    "latitude": 51.57173,
    "longitude": 0.21013,
    "name": "Hillview Food and Wine",
  },
  {
    "id": "UK00012819",
    "latitude": 51.54379,
    "longitude": 0.20829,
    "name": "TESCO SUPERSTORE Hornchurch",
  },
  {
    "id": "UK00017834",
    "latitude": 51.57529,
    "longitude": 0.20549,
    "name": "Loco Sid's",
  },
  {
    "id": "UKBRE00001",
    "latitude": 51.57462,
    "longitude": 0.31358,
    "name": "Basildon Petrol Station",
  },
  {
    "id": "UK00012973",
    "latitude": 51.56634,
    "longitude": 0.19188,
    "name": "TESCO Romford Extra",
  },
]
`;

exports[`findLocationsByPostcode throws an error for a non-200 response 1`] = `[Error: Post code not found.]`;

exports[`getAvailabilityForLocation returns availability data for a location 1`] = `
{
  "availabilityByLockerSize": {
    "large": {
      "availableCount": 3,
      "totalCount": 5,
    },
    "medium": {
      "availableCount": 8,
      "totalCount": 16,
    },
    "small": {
      "availableCount": 8,
      "totalCount": 15,
    },
  },
  "lastUpdatedAt": 2022-11-27T18:01:16.000Z,
}
`;

exports[`getAvailabilityForLocation throws an error for a non-200 response 1`] = `[Error: Locker [UK00058] not found or no data available]`;
